lib/ansible/modules/identity/ipa/ipa_host.py

```diff
@@ -221,7 +245,7 @@ def ensure(module, client):
                     data = {}
                     for key in diff:
                         data[key] = module_host.get(key)
-                    client.host_mod(name=name, host=data)
+                    return changed, client.host_mod(name=name, host=data)
```

Comment:
There is an issue when the host has already been enrolled to IPA and we call the module with randompassword: true. The module will properly create a random password but it does not remove the existing keytab and calling ipa-join or ipa-client-install with the new OTP will fail (because the host already has a keytab).
A possible fix would be to call ipa host-disable in this case, right before host-mod.

Reply:
Make sense. I will fix it.